### PR TITLE
Map HTTP status codes into gRPC failures.

### PR DESCRIPTION
**Pull request by ignasi35**
References #1165 

This goes on top of https://github.com/akka/akka-grpc/pull/1178

### CODE REVIEW

**Comment by raboof** [comment]
https://github.com/akka/akka-grpc/pull/1182/commits/a3b4fa6cdaa17d5dc0c98e76571c9631d50390e0 looks pretty good. Does it need to be on top of #1178, or could it be PR'ed to `akka-http-client` on its own? #1178 needs some more work...

**Comment by ignasi35** [comment]
> [a3b4fa6](https://github.com/akka/akka-grpc/commit/a3b4fa6cdaa17d5dc0c98e76571c9631d50390e0) looks pretty good. Does it need to be on top of #1178, or could it be PR'ed to `akka-http-client` on its own? #1178 needs some more work...

I've rebased to move it away from #1178. It looks like it does need to be on top of #1178 😒

**Comment by raboof** [comment]
> > [a3b4fa6](https://github.com/akka/akka-grpc/commit/a3b4fa6cdaa17d5dc0c98e76571c9631d50390e0) looks pretty good. Does it need to be on top of #1178, or could it be PR'ed to `akka-http-client` on its own? #1178 needs some more work...
> 
> I've rebased to move it away from #1178. It looks like it does need to be on top of #1178 unamused

I don't think so. The flaky test failures on this branch are indeed the ones that #1178 are intended to fix, but they are already present on `akka-http-client`, so this PR can be merged independently of #1178.

File: runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@ import akka.annotation.InternalApi
 import akka.event.LoggingAdapter
 import akka.grpc.GrpcProtocol.GrpcProtocolReader
 import akka.grpc.{ GrpcClientSettings, GrpcResponseMetadata, GrpcSingleResponse, ProtobufSerializer }
-import akka.http.scaladsl.model.HttpEntity.{ Chunk, Chunked, LastChunk }
+import akka.http.scaladsl.model.HttpEntity.{ Chunk, ChunkStreamPart, Chunked, Default, LastChunk }
```
Comment (raboof) [inline runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala:17]:
```suggestion
import akka.http.scaladsl.model.HttpEntity.{ Chunk, Chunked, Default, LastChunk }
```

**Pull request merged**